cmd/logs.go

```diff
@@ -0,0 +1,307 @@
+`
+
+const logsFilterFlagDesc = `Filter job logs to include only those steps that
+match certain filtering criteria. Use --filter
+name=value pairs. Available filters are
+compute_backend, docker_img, status and step.
+`
+
+func newLogsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "logs",
+		Short: "Get workflow logs.",
+		Long:  logsDesc,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			token, _ := cmd.Flags().GetString("access-token")
+			if token == "" {
+				token = os.Getenv("REANA_ACCESS_TOKEN")
+			}
+			serverURL := os.Getenv("REANA_SERVER_URL")
+			workflow, _ := cmd.Flags().GetString("workflow")
+			if workflow == "" {
+				workflow = os.Getenv("REANA_WORKON")
+			}
+
+			if err := validation.ValidateAccessToken(token); err != nil {
+				return err
+			}
+			if err := validation.ValidateServerURL(serverURL); err != nil {
+				return err
+			}
+			if err := validation.ValidateWorkflow(workflow); err != nil {
+				return err
+			}
+			if err := logs(cmd, token, workflow); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	cmd.Flags().Bool("json", false, "Get output in JSON format.")
+	cmd.Flags().StringP("access-token", "t", "", "Access token of the current user.")
+	cmd.Flags().
+		StringP("workflow", "w", "", "Name or UUID of the workflow. Overrides value of REANA_WORKON environment variable.")
+	cmd.Flags().StringSlice("filter", []string{}, logsFilterFlagDesc)
+	cmd.Flags().Int64("page", 1, "Results page number (to be used with --size).")
+	cmd.Flags().Int64("size", 0, "Size of results per page (to be used with --page).")
+
+	return cmd
+}
+
+func logs(cmd *cobra.Command, token string, workflow string) error {
+	filters, _ := cmd.Flags().GetStringSlice("filter")
+	jsonOutput, _ := cmd.Flags().GetBool("json")
+	page, _ := cmd.Flags().GetInt64("page")
+	size, _ := cmd.Flags().GetInt64("size")
+
+	steps, chosenFilters, err := parseLogsFilters(filters)
+	if err != nil {
+		return err
+	}
+
+	logsParams := operations.NewGetWorkflowLogsParams()
+	logsParams.SetAccessToken(&token)
+	logsParams.SetWorkflowIDOrName(workflow)
+	logsParams.SetPage(&page)
+	logsParams.SetSteps(steps)
+	if cmd.Flags().Changed("size") {
+		logsParams.SetSize(&size)
+	}
+
+	api, err := client.ApiClient()
+	if err != nil {
+		return err
+	}
+	logsResp, err := api.Operations.GetWorkflowLogs(logsParams)
+	if err != nil {
+		return fmt.Errorf("workflow logs could not be retrieved:\n%v", err)
+	}
+
+	var workflowLogs Logs
+	err = json.Unmarshal([]byte(logsResp.GetPayload().Logs), &workflowLogs)
+	if err != nil {
+		return err
+	}
+
+	err = filterJobLogs(&workflowLogs.JobLogs, chosenFilters)
+	if err != nil {
+		return err
+	}
+
+	if jsonOutput {
+		err := utils.DisplayJsonOutput(workflowLogs, cmd.OutOrStdout())
+		if err != nil {
+			return err
+		}
+	} else {
+		displayHumanFriendlyLogs(cmd, workflowLogs, steps)
+	}
+
+	return nil
+}
+
+func parseLogsFilters(filters []string) ([]string, map[string]string, error) {
```

Comment:
I think this is worth exploring. In the python client, filters are a bit all over the place and they work slightly differently for different commands. I think it'd be nice to have a new way to standardize them in go. I'll open a issue for that, since it'll probably have big changes to the code structure